pkg/asset/installconfig/powervs/platform.go

```diff
@@ -1,132 +1,25 @@
-	//                 need need to be different based on powervs's implementation of knownRegions
-	//
-
-	longRegions := make([]string, 0, len(regions))
-	shortRegions := make([]string, 0, len(regions))
-	for id, location := range regions {
-		longRegions = append(longRegions, fmt.Sprintf("%s (%s)", id, location))
-		shortRegions = append(shortRegions, id)
-	}
-
-	var regionTransform survey.Transformer = func(ans interface{}) interface{} {
-		switch v := ans.(type) {
-		case core.OptionAnswer:
-			return core.OptionAnswer{Value: strings.SplitN(v.Value, " ", 2)[0], Index: v.Index}
-		case string:
-			return strings.SplitN(v, " ", 2)[0]
-		}
-		return ""
-	}
 
 	ssn, err := GetSession()
 	if err != nil {
 		return nil, err
 	}
 
-	var region string
-
-	sessionRegion := ssn.Session.Region
-	if sessionRegion != "" {
-		if IsKnownRegion(sessionRegion) {
-			region = sessionRegion
-		} else {
-			logrus.Warnf("Unrecognized Power VS region %s, ignoring IC_REGION", sessionRegion)
-		}
-	}
-
-	sort.Strings(longRegions)
-	sort.Strings(shortRegions)
-	if region == "" {
-		err = survey.Ask([]*survey.Question{
-			{
-				Prompt: &survey.Select{
-					Message: "Region",
-					Help:    "The Power VS region to be used for installation.",
-					// Default: fmt.Sprintf("%s (%s)", defaultRegion, regions[defaultRegion]),
-					Options: longRegions,
-				},
-				Validate: survey.ComposeValidators(survey.Required, func(ans interface{}) error {
-					choice := regionTransform(ans).(core.OptionAnswer).Value
-					i := sort.SearchStrings(shortRegions, choice)
-					if i == len(shortRegions) || shortRegions[i] != choice {
-						return errors.Errorf("invalid region %q", choice)
-					}
-					return nil
-				}),
-				Transform: regionTransform,
-			},
-		}, &region)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	zones := knownZones(region)
-	defaultZone := zones[0]
-
-	var zoneTransform survey.Transformer = func(ans interface{}) interface{} {
-		switch v := ans.(type) {
-		case core.OptionAnswer:
-			return core.OptionAnswer{Value: strings.SplitN(v.Value, " ", 2)[0], Index: v.Index}
-		case string:
-			return strings.SplitN(v, " ", 2)[0]
-		}
-		return ""
-	}
-
-	var zone string
-	err = survey.Ask([]*survey.Question{
-		{
-			Prompt: &survey.Select{
-				Message: "Zone",
-				Help:    "The powervs zone within the region to be used for installation.",
-				Default: fmt.Sprintf("%s", defaultZone),
-				Options: zones,
-			},
-			Validate: survey.ComposeValidators(survey.Required, func(ans interface{}) error {
-				choice := zoneTransform(ans).(core.OptionAnswer).Value
-				i := sort.SearchStrings(zones, choice)
-				if i == len(zones) || zones[i] != choice {
-					return errors.Errorf("invalid zone %q", choice)
-				}
-				return nil
-			}),
-			Transform: zoneTransform,
-		},
-	}, &zone)
-	if err != nil {
-		return nil, err
-	}
-
 	var p powervs.Platform
-	if osOverride := os.Getenv("OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE"); len(osOverride) != 0 {
```

Comment:
this check should stay here. am i missing some technical reason why you moved it?

Reply:
A few meetings ago, I believe you said that all of the os.Getenv and survey questions were located in session.go.  When I raised that platform.go had a bunch you seemed confused but accepted it.  As this is the last variable in this file that required the os package for os.Getenv I just moved it to session.go along with the others.

I suppose I also don't understand what version of session you are referring to above.  As the variables are coming out of the users current environment, I read session as the current environment.  If I pulled another shell I'd get another environment with different variables.  I guess I was using environment and session interchangeably.

I am guessing session in this case refers to an ibmcloud session or a powervs session or something.  Probably and ibmpisession object... which begs the question why aren't we returning just the ibmpisession object in that case, and why we would be ok carrying the Creds object along side the session object.

Idk... I can put this variable wherever.

Reply:
I figure if we are keeping os image override in platform.go we can move all the region/zone stuff to region.go and call it from session.go.  Would that make more sense?

Reply:
> I read session as the current environment.

Now I see why these have all been put into that struct. That wasn't my use of this.

>Probably an ibmpisession object.

Yep

> why aren't we returning just the ibmpisession object in that case,

I can't remember why I didn't do this. Something about it didn't work, but I agree that it does seem to make more sense on paper to return it instead.

> I figure if we are keeping os image override in platform.go we can move all the region/zone stuff to region.go and call it from session.go. Would that make more sense?

Yah that sounds good